core/shared/src/main/scala/org/http4s/Media.scala

```diff
@@ -47,36 +39,50 @@ trait Media[F[_]] {
   final def charset: Option[Charset] =
     contentType.flatMap(_.charset)
 
-  // Decoding methods
-
-  /** Decode the [[Media]] to the specified type
-    *
-    * @param decoder [[EntityDecoder]] used to decode the [[Media]]
-    * @tparam T type of the result
-    * @return the effect which will generate the `DecodeResult[T]`
-    */
-  final def attemptAs[T](implicit decoder: EntityDecoder[F, T]): DecodeResult[F, T] =
-    decoder.decode(this, strict = false)
-
-  /** Decode the [[Media]] to the specified type
-    *
-    * If no valid [[Status]] has been described, allow Ok
-    *
-    * @param decoder [[EntityDecoder]] used to decode the [[Media]]
-    * @tparam A type of the result
-    * @return the effect which will generate the A
-    */
-  final def as[A](implicit F: MonadThrow[F], decoder: EntityDecoder[F, A]): F[A] =
-    F.rethrow(attemptAs.value)
 }
 
 object Media {
+
+  implicit final class InvariantOps[F[_]](private val self: Media[F]) extends AnyVal {
+    import self._
+
+    def bodyText(implicit
+        RT: RaiseThrowable[F],
+        defaultCharset: Charset = `UTF-8`,
+    ): Stream[F, String] = {
+      val cs = charset.getOrElse(defaultCharset).nioCharset
+      body.through(decodeWithCharset(cs))
+    }
+
+    // Decoding methods
+
+    /** Decode the [[Media]] to the specified type
+      *
+      * @param decoder [[EntityDecoder]] used to decode the [[Media]]
+      * @tparam T type of the result
+      * @return the effect which will generate the `DecodeResult[T]`
+      */
+    def attemptAs[T](implicit decoder: EntityDecoder[F, T]): DecodeResult[F, T] =
```

Comment:
Does the `F2` thing work for this one?

Reply:
Not this one, as it’s intended to be called with exactly one explicit type argument (e.g. attemptAs[String]). But the ones with no type argument, yes.

Reply:
Ugly, but this seems like a case for http://typelevel.org/cats/guidelines.html#-partially-applied-type.

Reply:
I don’t think that works without an explicit value argument?

Reply:
Oh yeah. Probably not :(